pkg/store/configuration/watchers.go

```diff
@@ -0,0 +1,69 @@
+// SPDX-FileCopyrightText: ${year}-present Open Networking Foundation <info@opennetworking.org>
+// SPDX-License-Identifier: LicenseRef-ONF-Member-Only-1.0
+
+package configuration
+
+import (
+	"sync"
+
+	"github.com/google/uuid"
+	configapi "github.com/onosproject/onos-api/go/onos/config/v2"
+)
+
+type watcher struct {
+	eventCh chan<- configapi.ConfigurationEvent
+	replay  bool
+	watchID configapi.ConfigurationID
+}
+
+// watchers stores the information about watchers
+type watchers struct {
+	watchers map[uuid.UUID]watcher
+	rm       sync.RWMutex
+}
+
+// newWatchers creates watchers
+func newWatchers() *watchers {
+	return &watchers{
+		watchers: make(map[uuid.UUID]watcher),
+	}
+}
+
+// sendAll sends a configuration event for all registered watchers
+func (ws *watchers) sendAll(event configapi.ConfigurationEvent) {
+	ws.rm.RLock()
+	go func() {
```

Comment:
We need to avoid creating a new goroutine for each event because it can cause the events to be reordered. Even though the data structure does not guarantee any order across keys, events _are_ ordered with respect to each key. Two concurrent updates to the same key will produce two events, and the events will be received in the order in which those updates were applied to the store. Handling those events on two different goroutines can cause them to be reordered.